src/tss2-esys/esys_crypto.c

```diff
@@ -1127,3 +1128,47 @@ iesys_cryptogcry_sym_aes_decrypt(uint8_t * key,
     gcry_cipher_close(cipher_hd);
     return TSS2_RC_SUCCESS;
 }
+
+
+TSS2_RC
+iesys_xor_parameter_obfuscation(TPM2_ALG_ID hash_alg,
+                                uint8_t *key,
+                                size_t key_size,
+                                TPM2B_NONCE * contextU,
+                                TPM2B_NONCE * contextV,
+                                BYTE *data,
+                                size_t data_size)
+{
+    TSS2_RC r;
+    uint32_t counter = 0;
+    BYTE  kdfa_result[TPM2_MAX_DIGEST_BUFFER];
+    size_t digest_size;
+    size_t data_size_bits = data_size * 8;
+    size_t rest_size = data_size;
+    BYTE *kdfa_byte_ptr;
+    r = iesys_crypto_hash_get_digest_size(hash_alg, &digest_size);
+    if (r != TSS2_RC_SUCCESS) {
```

Comment:
return_if_error(r, "Hash alg not supported");

Reply:
done